src/main/java/net/mcreator/generator/Generator.java

```diff
@@ -257,15 +258,37 @@ public List<GeneratorFile> generateElement(GeneratableElement element, boolean f
 		// generate all source files
 		List<GeneratorTemplate> generatorTemplateList = getModElementGeneratorTemplatesList(element.getModElement(),
 				performFSTasks, element);
+		List<GeneratorTemplatesList> templateLists = getModElementListTemplates(element.getModElement(), performFSTasks,
+				element);
 		if (generatorTemplateList != null) {
 			for (GeneratorTemplate generatorTemplate : generatorTemplateList) {
 				String templateFileName = (String) ((Map<?, ?>) generatorTemplate.getTemplateData()).get("template");
 
 				Map<String, Object> dataModel = generatorTemplate.getDataModel();
 				extractVariables(generatorTemplate, dataModel);
 
-				String code = getTemplateGeneratorFromName("templates").generateElementFromTemplate(element,
-						templateFileName, dataModel, element.getAdditionalTemplateData());
+				String code = "";
+				if (generatorTemplate.isListTemplate() && templateLists != null) {
+					boolean listItemFound = false;
+					for (GeneratorTemplatesList gtl : templateLists) {
+						for (int i = 0; i < gtl.listData().size(); i++) {
+							GeneratorTemplate listTemplate = gtl.getCorrespondingListTemplate(
+									generatorTemplate.getFile(), false);
+							if (listTemplate != null && gtl.processTokens(listTemplate, i).getAbsolutePath()
+									.equals(generatorTemplate.getFile().getAbsolutePath())) {
+								listItemFound = true;
+								code = getTemplateGeneratorFromName("templates").generateListItemFromTemplate(

```

Comment:
Code should be generated in getModElementListTemplates, not here

Reply:
Why? And why wasn't it before?

Reply:
Ok it is ok now I guess, I thought it would be better to generate in getModElementListTemplates to avoid double-lookup for GE. Consider resolved